packages-ts/starknet-gauntlet-ocr2/src/commands/ocr2/setConfig.ts

```diff
@@ -24,7 +26,58 @@ type ContractInput = [
   offchain_config: string[],
 ]
 
-const makeContractInput = async (input: SetConfigInput): Promise<ContractInput> => {
+const makeContractInput = async (input: SetConfigInput, ctx: ExecutionContext): Promise<ContractInput> => {
+  if (ctx.rdd) {
+    const contract = ctx.rdd[CONTRACT_TYPES.AGGREGATOR][ctx.contractAddress];
+    const config = contract.config;
+    const operators = contract.oracles.map((oracle) => ctx.rdd.operators[oracle])
+
+    const publicKeys = operators.map((o) => o.ocr2OffchainPublicKey[0].replace('ocr2off_terra_', ''))
+    const peerIds = operators.map((o) => o.peerId[0])
+    const operatorConfigPublicKeys = operators.map((o) =>
+      o.ocr2ConfigPublicKey[0].replace('ocr2cfg_terra_', ''),
+    )
+
+    const offchainConfig: encoding.OffchainConfig = {
+      deltaProgressNanoseconds: time.durationToNanoseconds(config.deltaProgress).toNumber(),
+      deltaResendNanoseconds: time.durationToNanoseconds(config.deltaResend).toNumber(),
+      deltaRoundNanoseconds: time.durationToNanoseconds(config.deltaRound).toNumber(),
+      deltaGraceNanoseconds: time.durationToNanoseconds(config.deltaGrace).toNumber(),
+      deltaStageNanoseconds: time.durationToNanoseconds(config.deltaStage).toNumber(),
+      rMax: config.rMax,
+      s: config.s,
+      offchainPublicKeys: publicKeys,
+      peerIds: peerIds,
+      reportingPluginConfig: {
+        alphaReportInfinite: config.reportingPluginConfig.alphaReportInfinite,
+        alphaReportPpb: Number(config.reportingPluginConfig.alphaReportPpb),
+        alphaAcceptInfinite: config.reportingPluginConfig.alphaAcceptInfinite,
+        alphaAcceptPpb: Number(config.reportingPluginConfig.alphaAcceptPpb),
+        deltaCNanoseconds: time.durationToNanoseconds(config.reportingPluginConfig.deltaC).toNumber(),
+      },
+      maxDurationQueryNanoseconds: time.durationToNanoseconds(config.maxDurationQuery).toNumber(),
+      maxDurationObservationNanoseconds: time.durationToNanoseconds(config.maxDurationObservation).toNumber(),
+      maxDurationReportNanoseconds: time.durationToNanoseconds(config.maxDurationReport).toNumber(),
+      maxDurationShouldAcceptFinalizedReportNanoseconds: time
+        .durationToNanoseconds(config.maxDurationShouldAcceptFinalizedReport)
+        .toNumber(),
+      maxDurationShouldTransmitAcceptedReportNanoseconds: time
+        .durationToNanoseconds(config.maxDurationShouldTransmitAcceptedReport)
+        .toNumber(),
+      configPublicKeys: operatorConfigPublicKeys,
+    };
```

Comment:
This is a big blob that should be part of gauntlet-core. Duplicated on other chains